server/athenian/api/internal/miners/jira/issue.py

```diff
@@ -864,12 +944,29 @@ def query_finish(query):
         query = [query_finish(query_start().where(*and_filters))]
 
     def hint_athenian_issue(q):
-        return (
-            q.with_statement_hint("Leading(((athenian_issue issue) (s c)))")
-            .with_statement_hint("Rows(athenian_issue issue *1000)")
-            .with_statement_hint("HashJoin(athenian_issue issue)")
-            .with_statement_hint("Rows(s c *200)")
-        )
+        AthenianIssueT = AthenianIssue.__tablename__
+        IssueT = Issue.__tablename__
+        # "s" and "c" are table aliases used in the api_statuses view
+        hints = [
+            f"Rows({AthenianIssueT} {IssueT} *1000)",
+            f"HashJoin({AthenianIssueT} {IssueT})",
+            "Rows(s c *200)",
+        ]
+        if mapped_to_prs is not None:
+            PRIssuesT = NodePullRequestJiraIssues.__tablename__
+            if len(mapped_to_prs) > _PR_ISSUES_THRESHOLD:
+                hints.append(
+                    f"Leading((((({PRIssuesT} *VALUES*) {AthenianIssueT}) {IssueT}) (s c)))",
+                )
+                hints.append(f"HashJoin({PRIssuesT} *VALUES*)")
```

Comment:
Unless Postgres is completely insane, this already happens without a hint. I've never seen it not choosing the HashJoin in practice.
What I completely forgot about is the row counts. Row counts can really backfire. We should add
```
Rows({PRIssuesT} *VALUES* #{len(mapped_to_prs)})
```

Reply:
Please share your query plan.

Reply:
with HashJoin
https://explain.tensor.ru/archive/explain/273dc4450b58d0d8010f7d6b387b1ae1:0:2023-03-16
without
https://explain.tensor.ru/archive/explain/c0b10d0e0f787a738fc6c883632ab8bf:0:2023-03-16

Reply:
OK, this is convincing... IDK why it doesn´t choose a HasHJoin automatically, TBH.
Let's add Rows and finish with it.

Reply:
added